torch/nn/parallel/distributed.py

```diff
@@ -646,8 +644,8 @@ def __init__(
             )
 
         # Check that a module does not have Uninitialized parameters
-        for param in module.parameters():
-            if isinstance(param, torch.nn.parameter.UninitializedParameter):
+        for name, param in module.named_parameters():
+            if isinstance(param, torch.nn.parameter.UninitializedParameter) and name not in self.parameters_to_ignore:
```

Comment:
Let's add a unittest where we have a parameter that's ignored and it is unitialized? 

Also, this is technically BC breaking, how do we want to deal with that?

Reply:
sure, I can add a test showing it breaking with ShardedTensor when ignored  today, and this fixes

re BC breaking -> IMO this is bug fix, not BC breaking functionality.